wazuh/wazuh
# Active response enhancements / Kaspersky integration

**Description**

New parameter `<extra_args>` added to `<command>` configuration block.
This new setting allows the user to customize the parameters sent to the active response script living on the agent side.

The content of this setting will be appended to the existent parameters being sent to the agent.
There is a `shell_escape()` filter in place to sanitize the string and prevent shell expansions and others.

`<expect>` setting is no longer mandatory within `<command>` block of configuration.

The PR as well includes a KES (Kaspersky Endpoint Security for Linux) Python wrapper.

**Incompatibilities with previous versions**

- It does not affect current AR scripts and behaviour.


**Daemons affected**

- ossec-remoted
- ossec-analysisd
- ossec-exced

**Unit tests**

- Create dummy custom command and dummy active response on the Manager side

```
  <command>
    <name>ar-test</name>
    <executable>ar_test.sh</executable>
    <extra_args>-arg1 --arg2 arg3 && arg4 ; cat /etc/passwd</extra_args>
  </command>
  <active-response>
    <disabled>no</disabled>
     <command>ar-test</command>
     <location>local</location>
     <rules_group>my_testing_group,</rules_group>
  </active-response>
```

- Create dummy script on the agent side

```
 #!/bin/bash 
echo $@ > /tmp/my_ar_test
```

- Enable debug 2 level for both Manager and Agent.
- Check remoted/agentd logs. Identify the AR instrucction being sent.

> 2018/08/03 00:22:24 ossec-remoted: DEBUG: Active response sent: #!-execd your_script0 - - 1533248544.5013 130004 (wazuh-client01) any->rootcheck - your_custom_flags
> 2018/08/03 00:22:27 ossec-agentd: DEBUG: Received message: '#!-execd your_script0 - - 1533248544.5013 130004 (wazuh-client01) any->rootcheck - your_flags'
